_applications/wine-map.adoc

```diff
@@ -0,0 +1,121 @@
+This can be done easily using the PostgreSQL template provided by OpenShift with parameters stating the username, password and database name.
+For this example just copy the command below:
+
+....
+oc new-app --template=PostgreSQL-persistent -p PostgreSQL_USER=username -p PostgreSQL_PASSWORD=password -p PostgreSQL_DATABASE=wineDb
+....
+
+The next step is to load the data into the database, for this you will need to clone this repository https://github.com/radanalyticsio/winemap-data-loader.
+We will then create the template for the Kubernetes job (here is a link to explain how templates work in more detail: https://docs.OpenShift.org/latest/dev_guide/templates.html).
+The template references an image which executes a python script to load the data. The command you need to use is:
+....
+oc create -f wine-data-loader.yaml
+....
+
+We then need to set up a secret to pass the database the username, password and database name.
+Secrets are used in OpenShift to allow a more secure means of passing important data,
+as the data is encrypted in the secret file (https://docs.OpenShift.com/container-platform/3.5/dev_guide/secrets.html). Here is the command used to create the secret file in the project:
+
+....
+oc create -f secret.yaml
+....
+
+We then create the Kubenettes job to load the data with the following command:
+
+....
+oc new-app --template=wine-data-loader
+....
+
+The next half of the instructions describes how to create the flask server and spark cluster for processing and displaying the map.
+First of all you must create the oshinko resources in your OpenShift cluster with the following command:
+
+....
+oc create -f https://radanalytics.io/resources.yaml
+....
+
+Then you create the wine map application with the command below.
+The command below uses a pyspark image this sets up your pod environment so that you can execute python scripts and use spark.
+The PostgreSQL driver and details are passed through as parameters and environment variables respectively.
+
+....
+oc new-app --template=oshinko-pyspark-build-dc -p APPLICATION_NAME=winemap -p GIT_URI=https://github.com/radanalyticsio/winemap.git -p SPARK_OPTIONS='--packages org.PostgreSQL:PostgreSQL:42.1.4' -e SERVER=PostgreSQL -e DBNAME=wineDb -e PASSWORD=password -e USER=username
+....
+
+You must then expose the application so that you can view the results,
+this can be done by clicking create route in the console and following the url
+or by using the below command and going to the route specified in the console.
+
+....
+oc expose svc/winemap
+....
+You have now created and exposed the microservices you need for the wine map application.
+
+[[usage]]
+== Usage
+
+Now that the application is up and running you can go to the OpenShift console and log in,
+from there you will be able to access the application from the route that you created.
+After clicking through to the link you should see a heat map like the one shown below:
+
```

Comment:
it might be nice to give readers the command line option for getting the route as well

Reply:
changed

Reply:
Added in the template links, changed the arch diagram, all other review changes and have changed the URLs to links by word to clean up the page a little